app/src/main/java/org/fossasia/phimpme/gallery/activities/LFMainActivity.java

```diff
@@ -3217,6 +3180,64 @@ protected void onPostExecute(Void result) {
         }
     }
 
+    /* AsyncTask for Add to favourites operation */
+    private class AddToFavourites extends AsyncTask<Void, Integer, Integer>{
+
+        @Override
+        protected void onPreExecute() {
+            getNavigationBar();
+            swipeRefreshLayout.setRefreshing(true);
+            super.onPreExecute();
+        }
+
+        @Override protected Integer doInBackground(Void... voids) {
+            int count = 0;
+            realm = Realm.getDefaultInstance();
```

Comment:
Isn't this already initialized?

Reply:
@mayank I guess realm objects can only be accessed on the thread they were created and this is a background thread so in order perform realm operation we need to get an instance of realm in this thread. Same goes for every other background thread using realm.

Reply:
Why not use a local instance then?

Reply:
I think @mayank8318 is right. See this link - https://stackoverflow.com/questions/407255/difference-between-declaring-variables-before-or-in-loop . The answer says that don't leave a narrow gap between declaration and initialization.

Reply:
@mayank8318 @raj10071997 Yeah that can be done, will make the change.

Reply:
@mayank8318 @raj10071997  I don't think this is necessary since the realm object has been used quite a few times in LFMainActivity for the purpose of operating with favourites part, so I guess it'd be better if a Realm object is declared at the top(like it is now) and every time a realm instantiation is required this object can be used.